### PR TITLE
WW-995: Use common Header Body Footer components

### PR DESCRIPTION
**Pull request by certara-yaroslavivanov**

### CODE REVIEW

File: src/AngularWebWinNonlinCloud/ClientApp/src/app/app.component.ts [hunk 1]
```diff
@@ -1,79 +1,97 @@
...
+  private _subscriptions: Subscription[] = [];
 
-    constructor(
-        readonly authService: AuthService,
-        readonly ga: GoogleAnalyticsService,
-        private logger: AuthenticatedLoggingService,
-        private readonly _projectService: ProjectService,
-        private readonly _oktaApiService: OktaApiService) {
-          logger.buildPublishers();
+  private footerLinks: FooterLink[] = [
```
Comment (ankitsinghcertara) [inline src/AngularWebWinNonlinCloud/ClientApp/src/app/app.component.ts:22]: _footerLinks

File: src/AngularWebWinNonlinCloud/ClientApp/src/app/app.component.ts [hunk 2]
```diff
@@ -1,79 +1,97 @@
...
-          const getUserInfo = async (userId: string) => {
-            const userInfo = await this._oktaApiService.getUserInfo(userId).toPromise();
-            const firstName = _.get(userInfo, 'profile.firstName');
-            const lastName = _.get(userInfo, 'profile.lastName');
-            if (firstName && lastName) {
-              this.userName = `${firstName} ${lastName}`;
-            }
-          };
+  constructor(
+    readonly authService: AuthService,
```
Comment (ankitsinghcertara) [inline src/AngularWebWinNonlinCloud/ClientApp/src/app/app.component.ts:35]: access specifier even if public

File: src/AngularWebWinNonlinCloud/ClientApp/src/app/app.component.ts [hunk 3]
```diff
@@ -1,79 +1,97 @@
...
+        subscription.unsubscribe();
       }
+    });
+  }
+
+  public async logout(): Promise<any> {
+    await this.authService.logout();
+  }
+
+  private tryAndSetCurrentProjectName(): void {
```
Comment (ankitsinghcertara) [inline src/AngularWebWinNonlinCloud/ClientApp/src/app/app.component.ts:90]: lets set this method name to 'setCurrentProjectName'

File: src/AngularWebWinNonlinCloud/ClientApp/src/app/app.component.ts (hunk 1 above)
Comment (ankitsinghcertara) [reply src/AngularWebWinNonlinCloud/ClientApp/src/app/app.component.ts:22]: Fixed myself

File: src/AngularWebWinNonlinCloud/ClientApp/src/app/app.component.ts (hunk 2 above)
Comment (ankitsinghcertara) [reply src/AngularWebWinNonlinCloud/ClientApp/src/app/app.component.ts:35]: Fixed myself

File: src/AngularWebWinNonlinCloud/ClientApp/src/app/app.component.ts (hunk 3 above)
Comment (ankitsinghcertara) [reply src/AngularWebWinNonlinCloud/ClientApp/src/app/app.component.ts:90]: Fixed

File: src/AngularWebWinNonlinCloud/ClientApp/src/app/app.component.ts (hunk 1 above)
Comment (certara-yaroslavivanov) [reply src/AngularWebWinNonlinCloud/ClientApp/src/app/app.component.ts:22]: I will add them as the rules to tslint

**Pull request merged**